src/sentry/static/sentry/app/views/organizationMonitors/monitorHeaderActions.jsx

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {browserHistory} from 'react-router';
+
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {addErrorMessage, addLoadingMessage, removeIndicator} from 'app/actionCreators/indicator';
+import {logException} from 'app/utils/logging';
+import SentryTypes from 'app/sentryTypes';
+import Button from 'app/components/button';
+import Confirm from 'app/components/confirm';
+
+export default class MonitorHeaderActions extends React.Component {
+  static propTypes = {
+    monitor: SentryTypes.Monitor.isRequired,
+    orgId: PropTypes.string.isRequired,
+    onUpdate: PropTypes.func,
+  };
+
+  handleDelete = () => {
+    const {orgId, monitor} = this.props;
+    const redirectPath = `/organizations/${orgId}/monitors/`;
+    addLoadingMessage(t('Deleting Monitor...'));
+
+    new Client().requestPromise(
```

Comment:
If you use `withApi` HoC on `MonitorHeaderActions`, you'll get instance of `Client` in props and it'll operate the same as the `ApiMixin` where it cancels requests when we unmount the component.